Accounts and Transactions

IvanLuchkin opened the pull request:
- Added Account and Transaction entities
- Added corresponding repository and service layers
- Added integration tests
- Renaming in Role entity
- Minor fixes in UserRepo and RoleRepo tests

IvanLuchkin committed:
Accounts and Transactions

- Added Account and Transaction entities
- Added corresponding repository and service layers
- Added integration tests
- Renaming in Tole entity
- Minor fixes in UserRepo and RoleRepo tests

KseniiaMakarova commented on src/main/java/com/banking/bankservice/service/impl/AccountServiceImpl.java:
```diff
@@ -0,0 +1,67 @@
+        Account accountInDb = getByAccountNumber(accountNumber);
+        accountInDb.setActive(false);
+        accountRepository.save(accountInDb);
+    }
```
представь, что у тебя расширится функционал, и нужно будет с контроллера не только заблокировать юзера, но и обратно сделать активным, или будет отдельный ендпойнт для изменения емейла юзеру

Тогда тебе придется каждый раз дописывать метод для сервиса
Лучше дизайнить api (слой сервиса) таким образом, чтобы были многофункциональные методы, которые клиент может использовать по нескольку раз

поэтому мне кажется, лучше поместить логику account.setActive(false); в контроллер, и потом вызывать accountService.save()

KseniiaMakarova commented on src/main/java/com/banking/bankservice/service/impl/AccountServiceImpl.java:
```diff
@@ -0,0 +1,67 @@
+
+    @Override
+    public List<Transaction> getPaymentHistory(String accountNumber, int page, int size) {
+        return accountRepository.getPaymentHistory(accountNumber, PageRequest.of(page, size));
```
we should add sorting by date desc

IvanLuchkin committed:
Improvements

- sorting added in AccountRepo.getPaymentHistory()
- embedded mongo dependency removed
- aggregation pipeline test fixed
- AccountRepositoryTest refactored a bit
- add() and update() methods added to AccountService

KseniiaMakarova commented on src/main/java/com/banking/bankservice/service/impl/AccountServiceImpl.java:
```diff
@@ -33,10 +38,8 @@ public Account getByAccountNumber(String accountNumber) {
-        Account accountInDb = getByAccountNumber(accountNumber);
-        accountInDb.setActive(false);
-        accountRepository.save(accountInDb);
+    public Account update(Account account) {
```
to me it is disputable if we should have methods with different names but identical implementation

`save` operation is designed to both create and update

IvanLuchkin replied:
I have decided to add it just for semantic purposes. I could name it `saveOrUpdate()`
